### PR TITLE
 Changes for automatic documentation using jsongenerator

### PR DESCRIPTION
**Pull request by nkkrishnan2512**
This PR contains the changes for automatic documentation for wpeframework plugins using jsongenerator.This adds a function
 DocGenerator() which contains the logic to generate document automatically and installs it to image.
 This PR also contains option for including the cpp interface headerfile path that will substitute the {cppinterfacedir} tag
 in JsonGeneratortool command for document generation.

### CODE REVIEW

**Review by sebaszm — CHANGES_REQUESTED** [review]
I find this iteration better, less code is usually better, but some things are still missing for this to operate properly.

**Review by bramoosterhuis — COMMENTED** [review]
@nkkrishnan2512 In addition to @sebaszm's review: 

If you are planning to add documentation to the image, please make sure to check that the ```${CMAKE_INSTALL_PREFIX}/share/doc``` is added to the ```${PN}-doc``` package on bitbake/yocto _before_ rolling this out to plugins. It should be an automatic rule, but please double-check if our recipes handle this properly. None of the documentation should end up in the ```${PN}-dev``` or main packages. I know with Buildroot, the default documentation locations are wiped from the final image anyhow.

Maybe it would also be good to add a high-level option like ```BUILD_DOCS``` in [project.cmake.in](https://github.com/rdkcentral/Thunder/blob/master/cmake/project.cmake.in) and set it to ```ON``` by default. The DocGenerator function should check if the execution of the generation and installation logic is required.  I think parsing the arguments should always be done. This way, you can always decide explicitly opt-out for documentation generation and installation if you want. 

Explicitly opt-out will be mainly beneficial not to slow down short develop-build-test cycles while developing.

**Comment by rjpdasilva** [comment]
@bramoosterhuis,

For the moment, this change is only targeting the tool itself, not the recipes.
The change is being done in a way that any plugins wanting to have their documentation generated will have to find the tools package and then invoke DocGenerator() from their cmake files. As you mention, the `/share/doc` is already part of the `${PN}-doc` by default, and we have confirmed this.

Later on, we can think of ways to make this generic to all plugins and have some top-level variables, cmake files, recipe include files, etc. to have a simpler way to enable/disable automatic doc generation for plugins. In our side, we're currently only this for some specific plugins. Nevertheless, these proposed changes for now are just for having the basis and common ground to abstract the automatic documentation generation via the DogGenerator() cmake function. I agree that there should be as many arguments as required for making this generic, but for now, we'd prefer not focusing on the recipes side.

File: Tools/cmake/FindJsonGenerator.cmake.in [hunk 1]
```diff
@@ -106,3 +110,12 @@ function(JsonGenerator)
 endfunction(JsonGenerator)
 
 message(STATUS "JsonGenerator ready ${JSON_GENERATOR}")
+
+function(DocGenerator)
+
+    set(GENERATOR_SEARCH_PATH ${CMAKE_SYSROOT}${CMAKE_INSTALL_PREFIX}/include/${NAMESPACE})
+    set(CPPIFDIR_PATH ${CMAKE_SYSROOT}${CMAKE_INSTALL_PREFIX}/include/WPEFramework/interfaces/)
```
[the remark this answers is not in the document]
Comment (nkkrishnan2512) [reply Tools/cmake/FindJsonGenerator.cmake.in:117]: Okay

File: Tools/cmake/FindJsonGenerator.cmake.in [hunk 2]
```diff
@@ -106,3 +110,12 @@ function(JsonGenerator)
...
+    JsonGenerator(INPUT ${CMAKE_CURRENT_SOURCE_DIR}/${PLUGIN_NAME}Plugin.json DOCS OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/doc" INCLUDE_PATH ${GENERATOR_SEARCH_PATH} CPPIFDIR ${CPPIFDIR_PATH})
```
[the remark this answers is not in the document]
Comment (nkkrishnan2512) [reply Tools/cmake/FindJsonGenerator.cmake.in:118]: Ok

**Comment by nkkrishnan2512** [comment]
Raised new PR(https://github.com/rdkcentral/Thunder/pull/729) with review comment fix changes,hence closing this.

**Pull request closed without merging**